### ISSUE TITLE
Notepad++ v8.4.2 CRASHES or doesn't work properly when applying a color on selected text using NppExec

### ISSUE THREAD

**Issue by jabal1** [notepad-plus-plus/notepad-plus-plus#11829]
### Notepad++ v8.4 and beyond (inclusive 8.4.2) CRASHES or doesn't work properly when applying a color on selected text using NppExec.

In notepad++v8.3.3 everything works fine but when I wanted to upgrade to notepad++v8.4 and beyond Notepad++ went **unstable** and did not work as usual.

I always use two views, meaning I have text files open on two columns.
I want to have my selected text to be "GREEN on BLACK", therefore I have implemented NppExec with following script :

**SCI_SENDMSG 2067 1 65280**

This sets globally the foreground color to green fro selected text.
Although this initialization only works for one of the views (the active one at startup time) when I start notepad++, thus I always have also to apply it manually on the second view which does not have the focus at startup.

Now this worked perfectly well in v8.3.3 where I could have both views with selected text displayed "GREEN on BLACK".

But when I updated to v8.4 or beyond this did not work anymore.
I still had to execute the second NppExec command manually on the second view, but each time I lose the focus on one view the selected text returns to the default higlighting and more problematic notepad++ now very often just **CRASHES** when doing so !

### Steps to Reproduce the Issue

1. NppExec must be present
2. define a NppEXEC script used at startup containing--> **SCI_SENDMSG 2067 1 65280**
3. open notepad++having already 2 views with files open
4. execute manually the NppEexec script on the second view which had not the focus when notepad++was launched
5. change alternatively the focus from one view to another, when the focus leaves the view the corresponding selected text highlighting reverts to default ! Or even worse notepad++ simply **CRASHES** !

### Expected Behavior

As in v8.3.3 I expect the selected text to remain highlighted in both views in "GREEN on BLACK", and more importantly notepad++ should not **CRASH** as this happens very frequently with v8.4 and beyond !

### Actual Behavior

With v8.4 and beyond changing the focus to one view reverts the other view to the default highlighting for selected text, and it makes notepad++ **CRASH** very frequently.

**In fact it **CRASHES** so often I had to reinstall v8.3.3 !**

### Debug Information

NppExec script
![image](https://user-images.githubusercontent.com/65338289/175306032-75769dc4-88ba-41b9-bb13-ba55c84d557e.png)

**WORKING VERSION**
Notepad++ v8.3.3   (64-bit)
Build time : Mar 13 2022 - 17:20:02
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 2009
OS Build : 19044.1706
Current ANSI codepage : 1252
Plugins : DSpellCheck.dll mimeTools.dll NppConverter.dll NppExec.dll NppExport.dll NppHorizontalRuler.dll 

![image](https://user-images.githubusercontent.com/65338289/175305299-86629f7f-eb9f-4741-a1fe-374f63ef27a0.png)

**PROBLEMATIC VERSION**
Notepad++ v8.4   (64-bit)
Build time : Apr 20 2022 - 03:31:06
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 2009
OS Build : 19044.1706
Current ANSI codepage : 1252
Plugins : DSpellCheck.dll mimeTools.dll NppConverter.dll NppExec.dll NppExport.dll NppHorizontalRuler.dll 

![image](https://user-images.githubusercontent.com/65338289/175305541-6284c1f9-5089-4722-92a2-ee34e967cae0.png)

**Comment by alankilborn** [comment]
Strictly a guess, but I'd say that NppExec has to change for this aspect to work with N++ 8.4+.  Some reference is [HERE](https://community.notepad-plus-plus.org/topic/23146/faq-desk-notepad-crashes-freezes-unresponsive-after-update#:~:text=and%20a%20recent%20large%20Scintilla%20update%20in%20v8.4).

**Comment by alankilborn** [comment]
Suggest posting on NppExec support page, and it can be sorted out between that developer (and N++ devs if needed).

**Comment by jabal1** [comment]
I did a copy-paste of this Issue report there, I hope they will treat it, find the problem and fix it.

**Comment by jabal1** [comment]
Hello,

@rdipardo 

Thanks for your support, as suggested I have made the bug report for NppHorizontalRule.

Meanwhile I will also try the development build and give a feedback here when I can.

As for the "Use DirectWrite" option it did change nothing and whilst it says "need to restart Notepad++" (to be taken into account), it happens that the option is not persistent, at least in v8.4 when I restart notepad++ the option has gone. And when I enable it just for the session it doesn't seem to have an effect.

![image](https://user-images.githubusercontent.com/65338289/175483503-083c4921-efd3-4576-af37-bf37ae42c8ef.png)

So as soon as possible I will test the development build.

**Comment by jabal1** [comment]
@rdipardo

unfortunately the development build : 

https://ci.appveyor.com/project/donho/notepad-plus-plus/builds/43775785/job/i7yev8v0b8pc36fp/artifacts

does not change the situation, when losing the focus the highlighting of selected text reverts to the default.

![image](https://user-images.githubusercontent.com/65338289/175490080-730146e8-f6df-4d92-a04e-dbd2978259d6.png)

**Comment by alankilborn** [comment]
@jabal1 

The _ColumnTools_ plugin offers a similar horizontal ruler feature; you may want to give that one a try.

**Comment by alankilborn** [comment]
>>>The crash can be traced to NppHorizontalRuler

>>The ColumnTools plugin offers a similar horizontal ruler feature; you may want to give that one a try.

>maybe it's more evolved than the one I use which for instance has no option to change the ruler background which thus remains white.

I don't think it is more "evolved", but for me it is stable, and that is probably the primary feature you are currently looking for.  :-)

**Comment by alankilborn** [comment]
> Also I did not find any reliable website where to get ColumnTools.dll (for either x64 or x32) so despite my curiosity to try it out I guess that I will stick to NppHorizontalRuler for now.

See https://notepad-plus-plus.org/community/topic/17370/64-bit-plugins/
which leads to: https://github.com/vinsworldcom/nppColumnTools

**Comment by jabal1** [comment]
2 points.

1. Yes that's the place where I got the source code but there is no binary or dll to be downloaded from https://github.com/vinsworldcom/nppColumnTools. That was my problem !

2. I use the laptop of the office meaning Notepad++ does not natively launch in admin mode ! So the inbuild install/update does not work unless I start Notepad++.exe explicitly with administrative rights.

When I did it crashed, so I removed manually NppHorizontalRuler and then I could launch notepad++ as admin without crash. So yes definitely NppHorizontalRuler is not stable in certain circumstances with notepad 8.4.2.

Then I could go forward and install ColumnTools from notepad++'s inbuilt functionality, and it did not crash when being in admin mode, so yes it seems indeed more stable than NppHorizontalRuler.

They are not very different, but with NppHorizontalRuler you can define a smaller font so that the ruler is not as tall as the one in ColumnTools which remains at its default size.

In any case I thank you for showing me ColumnTools !

**Comment by alankilborn** [comment]
> that's the place where I got the source code but there is no binary or dll to be downloaded from https://github.com/vinsworldcom/nppColumnTools. That was my problem !

Hmm, binary is there; see https://github.com/vinsworldcom/nppColumnTools/releases/tag/1.4.4.1

**Comment by jabal1** [comment]
Right !

Naïvely I thought i'd find either a menu item with a dull "Download" tab

![image](https://user-images.githubusercontent.com/65338289/175555292-21b9e969-1d57-4048-bfd0-f1380a328806.png)

or a link with a text like "ColumnTools Download" or something, only after you showed me the link I started to click on

![image](https://user-images.githubusercontent.com/65338289/175555163-54bc05dd-9939-402a-860f-b9e69824f219.png)

I really expected a text like "ColumnTools" because seeing "**Notepad++ 8.4 / Scintilla 5+ up...**" under **Release** I just stupidly thought it was about downloading Notepad++ 8.4 and Scintilla 5+..., so I did not even click at this in first place...
Stupid I know... : D

Though I still think a dull "Download" tab would best serve the purpose : D

Anyway thanks

**Comment by rdipardo** [comment]
This can be closed as NppExec's maintainer has already [resolved the main issue](https://github.com/d0vgan/nppexec/issues/69).
